milli/src/search/criteria/proximity.rs

```diff
@@ -327,15 +327,25 @@ fn resolve_plane_sweep_candidates(
             let (_, (_, _, right_most_pos)) = groups.last()?;
 
             for pair in groups.windows(2) {
-                if let [(i1, (_, _, rpos1)), (i2, (lpos2, prox2, _))] = pair {
-                    // if a pair overlap, meaning that they share at least a word, we return None
-                    if rpos1 >= lpos2 { return None }
+                if let [(i1, (lpos1, _, rpos1)), (i2, (lpos2, prox2, rpos2))] = pair {
```

Comment:
Maybe you can put `&` before the name of the variable to avoid having to `*` deref them everywhere.

```suggestion
                if let [(i1, (&lpos1, _, &rpos1)), (i2, (&lpos2, &prox2, &rpos2))] = pair {
```